tests/testbed_setup/test_xr_migration.py

```diff
@@ -0,0 +1,210 @@
+    out2 = read_con(duthost_console, "EXIT_ON_SUCCESS: SONiC migration script exiting due to successful completion")
+    output = out + out2
+    if "[ERROR]" in output:
+        duthost_console.disconnect()
+        raise Exception("Error occured in rollback verification, exiting.")
+    sleep(5)
+
+    # Reconfigure the management IP
+    duthost_console.write_channel("config" + duthost_console.RETURN)
+    duthost_console.write_channel("line console" + duthost_console.RETURN)
+    duthost_console.write_channel("width 0" + duthost_console.RETURN)
+    duthost_console.write_channel("length 0" + duthost_console.RETURN)
+    duthost_console.write_channel(f"interface {mgmt_int}" + duthost_console.RETURN)
+    duthost_console.write_channel(f"ipv4 address {mgmt_ip_mask}" + duthost_console.RETURN)
+    duthost_console.write_channel("no shut" + duthost_console.RETURN)
+    duthost_console.write_channel("router static" + duthost_console.RETURN)
+    duthost_console.write_channel("address-family ipv4 unicast" + duthost_console.RETURN)
+    duthost_console.write_channel(f"0.0.0.0/0 {mgmt_gw}" + duthost_console.RETURN)
+    duthost_console.write_channel("ssh server v2" + duthost_console.RETURN)
+    duthost_console.write_channel("ssh server vrf default" + duthost_console.RETURN)
+    duthost_console.write_channel("commit" + duthost_console.RETURN)
+    duthost_console.write_channel("end" + duthost_console.RETURN)
+    logger.debug("after mgmt config")
+
+    # XR to Sonic Migration steps
+    duthost_console.timeout = 300
+    # Copy Required Files to Harddisk from migration file location
+    duthost_console.write_and_poll(f"scp {scp_user}@{scp_host}:{mig_file_path}* /harddisk:/", "Password:")
+    duthost_console.write_channel(scp_pass + duthost_console.RETURN)
+    read_con(duthost_console, xr_prompt)
+
+    # Run XR Upgrade Check
+    duthost_console.write_channel(f"run python /harddisk:/{mig_script} --xr_upgrade --check" + duthost_console.RETURN)
+    out = read_con(duthost_console, xr_prompt)
+    if "EXIT_ON_SUCCESS" not in out:
+        duthost_console.disconnect()
+        raise Exception("Error occured in XR upgrade check, exiting.")
+
+    # Upgrade to XR Migration Image
+    duthost_console.write_channel(f"run python /harddisk:/{mig_script} --xr_upgrade" + duthost_console.RETURN)
+    out = read_con(duthost_console, "XR_UPGRADE: Device will now reload for IOS XR upgrade to")
+    duthost_console.timeout = 2000
+    out2 = read_con(duthost_console, "Press RETURN to get started.")
+    output = out + out2
+    if "[ERROR]" in output:
+        duthost_console.disconnect()
+        raise Exception("An error happened during XR upgrade, exiting")
+    duthost_console.write_and_poll(duthost_console.RETURN, "Username:")
+    duthost_console.write_and_poll(console_user, "Password:")
+    duthost_console.write_and_poll(console_password, xr_prompt)
+    logger.debug("after migration image upgrade login")
+    sleep(120)
```

Comment:
Can we use wait_for instead of sleep for certain seconds?

Reply:
There is nothing to wait for at this time.  The script logs in, then gives the system more time to complete the boot up process so it is on the prompt and wont change in a deterministic fashion.